int32 operation
---
bool success
